packages/web/server/queries/complex/earn/strategies.ts

```diff
@@ -0,0 +1,122 @@
+import { Dec, PricePretty, RatePretty } from "@keplr-wallet/unit";
+import cachified, { CacheEntry } from "cachified";
+import { LRUCache } from "lru-cache";
+
+import { DEFAULT_LRU_OPTIONS } from "~/config/cache";
+import { type Asset, getAsset } from "~/server/queries/complex/assets";
+import { DEFAULT_VS_CURRENCY } from "~/server/queries/complex/assets/config";
+import {
+  EarnStrategy,
+  EarnStrategyBalance,
+  queryEarnStrategies,
+  queryEarnUserBalance,
+} from "~/server/queries/numia/earn";
+
+const earnStrategiesCache = new LRUCache<string, CacheEntry>(
+  DEFAULT_LRU_OPTIONS
+);
+
+const earnStrategyBalanceCache = new LRUCache<string, CacheEntry>(
+  DEFAULT_LRU_OPTIONS
+);
+
+export async function getEarnStrategies() {
+  return await cachified({
+    cache: earnStrategiesCache,
+    ttl: 1000 * 60 * 60,
+    key: "earn-strategies",
+    getFreshValue: async (): Promise<EarnStrategy[]> => {
+      try {
+        const strategies = await queryEarnStrategies();
+        const aggregatedStrategies: EarnStrategy[] = [];
+
+        for (let rawStrategy of strategies) {
+          const {
+            rewardDenoms,
+            tokenDenoms,
+            apy,
+            tvl,
+            id,
+            category,
+            lockDuration,
+            name,
+            provider,
+            type,
+            link,
+          } = rawStrategy;
+
+          const rewards = await Promise.all(
+            rewardDenoms.map((reward) =>
+              getAsset({ anyDenom: reward.symbol }).catch((_) => undefined)
+            )
+          );
+          const token = await Promise.all(
+            tokenDenoms.map((token) =>
+              getAsset({ anyDenom: token.symbol }).catch((_) => undefined)
+            )
+          );
+
+          const processedApy = new RatePretty(new Dec(apy));
+          const processedTvl = new PricePretty(
+            DEFAULT_VS_CURRENCY,
+            new Dec(tvl)
+          );
+
+          aggregatedStrategies.push({
+            id,
+            name,
+            category,
+            provider,
+            type,
+            involvedTokens: token.filter((reward) => !!reward) as Asset[],
+            rewardTokens: rewards.filter((reward) => !!reward) as Asset[],
+            lockDuration,
+            tvl: processedTvl,
+            apy: processedApy,
+            risk: 0,
+            balance: new PricePretty(DEFAULT_VS_CURRENCY, new Dec(0)),
+            hasLockingDuration: lockDuration > 0,
+            tokensType: "stablecoins", // todo
+            link,
+          });
+        }
+        return aggregatedStrategies;
+      } catch (error) {
+        return [];
```

Comment:
I think it would be good to allow errors to propagate for future debugging if there's issues fetching the strategies